checkmedia.c

```diff
@@ -464,22 +466,47 @@ void update_progress(unsigned size)
 }
 
 
-void check_mbr(unsigned char *mbr)
+/*
+ * Read 32 bit value at pointer address, little-endian.
+ */
+uint32_t read_le32(const void *ptr)
 {
-  unsigned char *p = mbr + 0x1be;	// partition table
-  unsigned s;
+  const uint8_t *s = ptr;
 
-  if(mbr[0x1fe] != 0x55 || mbr[0x1ff] != 0xaa) return;
+  return s[0] + (s[1] << 8) + (s[2] << 16) + (s[3] << 24);
+}
 
-  if(p[0] & 0x7f) return;
 
-  s = p[0x0c] + (p[0x0d] << 8) + (p[0x0e] << 16) + (p[0x0f] << 24);
+/*
+ * Analyze MBR to find image size.
+ *
+ * The result is stored in iso.hybrid_size;
+ */
+void check_mbr(const unsigned char *mbr)
+{
+  const uint8_t *p;
+  unsigned idx, image_size;
+  uint32_t start, length, end;
 
-  if(s <= 64) return;			// at least iso header
+  // check MBR signature
+  if(mbr[0x1fe] != 0x55 || mbr[0x1ff] != 0xaa) return;
+
+  /*
+   * Scan all 4 primary partition table entries. Find the maximum used
+   * address.
+   */
+  for(p = mbr + 0x1be, idx = 0, image_size = 0; idx < 4; p += 0x10, idx++) {
```

Comment:
I'm not an MBR expert, could please describe (comment) the magic constants used in in the code?